gems/aws-sdk-s3/lib/aws-sdk-s3/file_downloader.rb

```diff
@@ -94,7 +94,12 @@ def compute_chunk(file_size)
         if @chunk_size && @chunk_size > file_size
           raise ArgumentError, ":chunk_size shouldn't exceed total file size."
         else
-          @chunk_size || [(file_size.to_f / MAX_PARTS).ceil, MIN_CHUNK_SIZE].max.to_i
+          @chunk_size ||= [
```

Comment:
I don't think we want to actually set the `@chunk_size` as the object may be re-used.  Create a temp `computed_size`?